src/vs/base/common/async.ts

```diff
@@ -930,6 +930,86 @@ export class ResourceQueue implements IDisposable {
 	}
 }
 
+export type Task<T = void> = () => (Promise<T> | T);
+
+/**
+ * Processes tasks in the order they were scheduled.
+*/
+export class TaskQueue {
```

Comment:
@alexr00 @hediet maybe it would help to explain how this helper is different from `Queue`.